tools/snp.sh

```diff
@@ -164,6 +172,63 @@ verify_snp_host() {
   fi
 }
 
+verify_if_host_is_snp_capable() {
+  # Get the host cpuid eax
+  local host_cpuid_eax
+  host_cpuid_eax=$(cpuid -1 -r -l 0x8000001f | grep -oE "eax=[[:alnum:]]+ " | cut -c5- | tr -d '[:space:]')
+
+  # Bit 0 indicates support for SME status
+  local sme_bit=1
+  local sme_status=$((host_cpuid_eax & sme_bit))
+
+  #  Bit 1 indicates support for SEV 
+  local sev_bit=2
+  local sev_status=$((host_cpuid_eax & sev_bit))
+
+  # Bit 3 indicates support for SEV-ES
+  local sev_es_bit=8
+  local sev_es_status=$((host_cpuid_eax & sev_es_bit))
+
+  # Bit 4 indicates support for SNP
+  local snp_bit=16
+  local snp_status=$((host_cpuid_eax & snp_bit))
+
+  # Map all sev features in a single associative array for all SEV features support in the processor
+  declare -A all_actual_results=(
+    [SME]=${sme_status} 
+    [SEV]=${sev_status}
+    [SEV-ES]=${sev_es_status} 
+    [SNP]=${snp_status}
+  )
+
+  declare -A results_to_match=(
+    [SME]=1 
+    [SEV]=2
+    [SEV-ES]=8
+    [SNP]=16
+    )
```

Comment:
Is there a reason we are declaring this twice?

Reply:
I have done code cleanup for this